posts/2025-03-02-Rustup-1.28.0.md

```diff
@@ -0,0 +1,120 @@
+
+- rustup will no longer automatically install the active toolchain if it is not installed.
+
+  - To ensure its installation, run `rustup toolchain install` with no arguments.
+  - The following command installs the active toolchain both before and after this change:
+    ```sh
+    rustup show active-toolchain || rustup toolchain install
+    # Or, on older versions of PowerShell:
+    rustup show active-toolchain; if ($LASTEXITCODE -ne 0) { rustup toolchain install }
+    ```
+
+- Installing a host-incompatible toolchain via `rustup toolchain install` or `rustup default` will
+  now be rejected unless you explicitly add the `--force-non-host` flag.
+
+Rustup now officially supports the following host platforms:
+
+- `aarch64-pc-windows-msvc`
+- `loongarch64-unknown-linux-musl`
+
+This release also comes with various quality-of-life improvements, to name a few:
+
+- `rustup show`'s output format has been cleaned up, making it easier to find out about your toolchains' status.
+- `rustup doc` now accepts a flag and a topic at the same time, enabling quick navigation to specific parts of more books.
+- rustup's `remove` subcommands now support more aliases such as `rm` and `del`.
+- Basic support for nushell has been added.
+
+We have additionally made the following internal changes:
+
+- The default download backend has been changed from reqwest with native-tls to reqwest with rustls.
+  - `RUSTUP_USE_CURL` and `RUSTUP_USE_RUSTLS` can still be used to change the download backend
+    if the new backend causes issues. If issues do happen, please [let us know](https://github.com/rust-lang/rustup/issues/3806).
+  - The default backend now uses rustls-platform-verifier to verify server certificates, taking
+    advantage of the platform's certificate store on platforms that support it.
+- When creating proxy links, rustup will now try symlinks first and fall back to hardlinks,
+  as opposed to trying hardlinks first.
+- A new `RUSTUP_LOG` environment variable can be used to control tracing-based logging in
+  rustup binaries. See the [dev guide](https://rust-lang.github.io/rustup/dev-guide/tracing.html) for more details.
+
+Finally, there are some notable changes to our [official website][install] as well:
+
+- The overall design of the website has been updated to better align with the Rust Project's branding.
+- It is now possible to download the prebuilt `rustup-init.sh` installer for the `aarch64-pc-windows-msvc` host platform via https://win.rustup.rs/aarch64.
+
+Further details are available in the [changelog]!
+
+Rustup's documentation is also available in [the rustup book][book].
+
+[other installation methods]: https://rust-lang.github.io/rustup/installation/other.html
+[changelog]: https://github.com/rust-lang/rustup/blob/stable/CHANGELOG.md
+[book]: https://rust-lang.github.io/rustup/
+
+## Thanks
+
+Thanks again to all the contributors who made rustup 1.28.0 possible!
+
+- Alovchin91 (Alovchin91)
```

Comment:
I'm inclined to drop the list of contributors, maybe referring to the changelog instead.

Reply:
This corresponds to https://blog.rust-lang.org/2025/01/09/Rust-1.84.0.html's "Contributors to 1.84.0" section, but I have no particular preference WRT keeping this section as-is or delegate to the changelog.

Reply:
That doesn't inline the list of contributors either, though, right?

<img width="819" alt="Screenshot 2025-03-02 at 15 31 39" src="https://github.com/user-attachments/assets/d3260132-c6b6-4cfd-a204-845097105e48" />

Reply:
@djc The full changelog, I believe, is there in case someone wants to find out what have happened in this release cycle, whereas https://thanks.rust-lang.org gives a list of _contributors_ to express our gratitude instead of contributions, and I believe this is the main difference here. We should been mentioned in that website as well since a long time ago but alas... As I said I have no particular preference regarding what should be done right here & right now.

Reply:
rustup has a website so I wonder if we could add a thanks section somewhere there?

Reply:
Just a fly-by. I opened this: <https://github.com/rust-lang/thanks/issues/77>.

But yeah maybe rustup can publish it on its own website.